Checkout/Checkout/Source/Validation/Validators/CardNumberValidator.swift

```diff
@@ -8,8 +8,10 @@
 import Foundation
 
 public protocol CardNumberValidating {
+  typealias ValidationScheme = (isComplete: Bool, scheme: Card.Scheme)
   func eagerValidate(cardNumber: String) -> Result<Card.Scheme, ValidationError.EagerCardNumber>
   func validate(cardNumber: String) -> Result<Card.Scheme, ValidationError.CardNumber>
+  func validateCompleteness(cardNumber: String) -> Result<ValidationScheme, ValidationError.CardNumber>
```

Comment:
Very good feedback, thank you. I have now added documentations to the methods which hopefully would make it clearer. I am still not 100% cofident this is as clear as it could be so let me know if you have more ideas on how to improve them, I got too stuck in and can't see them anymore 😵

Reply:
@alex-ioja-yang-cko Can we not merge all these into one function validate ? and then internally call eagerValidate, validateCompleteness and logic for validate. Will that be a overhead ?. 

I was thinking to reduce the number of public methods so anyone using them finds it simple. Please correct me if I am wrong.

Reply:
I wanted to refactor `validate(cardNumber` into doing this and returning the new result. Spent a few hours on that path and kept having failing tests in Checkout, in old screen, everywhere...

My final solution was to stop wasting time on it and strip it out completely when deleting the old screen, as there will no longer be a consumer for it. Then we may be able to rename this as well. Actually I'll get this added to the relevant task now and let me know if you think that's sufficient or still want to address better 🤔

Reply:
I've created this [follow up task](https://checkout.atlassian.net/browse/PIMOB-1364) for myself, once the old UI is removed to remove old one and rename new one to that.

We'll end up again with a single validation method but which will just say yes/no (which is what we want, we never seemed to care about reasoning validation as that was done separately already).